### PR TITLE
DAOS-12964 control: Refine dmg hostlist help

### PR DESCRIPTION
**Pull request by kjacque**
- Associate -l/--host-list options only with commands that accept a host-list parameter.
- Deprecate and hide top-level version of the -l/--host-list option. This is to avoid breaking users' scripts if they put the -l option immediately after the dmg invocation. The value passed via this parameter is copied to the new one if it is a hostlist command.
- Add a single host version of the hostlist parameter with appropriate help text. For now this is used for telemetry commands.

Features: dmg telemetry

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by kjacque** [comment]
The failed rebuild EC test is a known issue: https://daosio.atlassian.net/browse/DAOS-12883

File: src/control/cmd/dmg/main.go [hunk 1]
```diff
@@ -62,8 +74,42 @@ func (cmd *ctlInvokerCmd) setInvoker(c control.Invoker) {
...
+	return []string{cmd.host}
+}
+
+func (cmd *singleHostCmd) validateHostList() error {
+	numHosts := cmd.HostList.Count()
+	if numHosts > 1 {
+		return fmt.Errorf("must pass in exactly 1 host (got %d)", numHosts)
+	}
+	return nil
+}
```
Comment (mjmac) [inline src/control/cmd/dmg/main.go:109]: This is an OK way to do it, but it feels a little funny to make the normal implementation use a stub. A slightly different approach would be to add a new local type alias for `ui.HostSetFlag` and give it a custom `UnmarshalFlag()` method, e.g.

```go
type singleHostFlag ui.HostSetFlag

func (shf *singleHostFlag) UnmarshalFlag(value string) error {
	if err := (*ui.HostSetFlag)(shf).UnmarshalFlag(value); err != nil {
		return err
	}

	if shf.Count() != 1 {
		return errors.New("must specify a single host")
	}

	return nil
}
```

Something to consider, not a required change, though.

**Review by mjmac — APPROVED** [review]
Looks good. Pretty clean solution for the legacy hostlist option, very nice.

File: src/control/cmd/dmg/main.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/dmg/main.go:109]: Good point. I like your suggestion, it feels cleaner to me.

**Review by mjmac — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by tanabarr — APPROVED** [review]

**Pull request merged**